sdk/core/core/src/az_http_header_validation_private.h

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_http_header_validation_private.h
+ *
+ * @brief This header defines a bit array that is used to validate whenever or not an ASCII char is
+ * valid within an http header name.
+ *
+ */
+
+#ifndef _az_HTTP_HEADER_VALIDATION_PRIVATE_H
+#define _az_HTTP_HEADER_VALIDATION_PRIVATE_H
+
+#include <az_span.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+// Bit array from ASCII to valid. Every zero in array means invalid character, while non-zeros
+// return the valid character.
+static const uint8_t az_http_valid_token[256] = {
+  0, // 0 ->  null
+  0, // 1 ->  start of heading
+  0, // 2 ->  start of text
+  0, // 3 ->  end of text
+  0, // 4 ->  end of transmission
+  0, // 5 ->  enquiry
+  0, // 6 ->  acknowledge
+  0, // 7 ->  bell
+  0, // 8 ->  backspace
+  0, // 9 ->  horizontal tab
+  0, // 10 ->  new line
+  0, // 11 ->  vertical tab
+  0, // 12 ->  new page
+  0, // 13 ->  carriage return
+  0, // 14 ->  shift out
+  0, // 15 ->  shift in
+  0, // 16 ->  data link escape
+  0, // 17 ->  device control 1
+  0, // 18 ->  device control 2
+  0, // 19 ->  device control 3
+  0, // 20 ->  device control 4
+  0, // 21 ->  negative acknowledge
+  0, // 22 ->  synchronous idle
+  0, // 23 ->  end of trans. block
+  0, // 24 ->  cancel
+  0, // 25 ->  end of medium
+  0, // 26 ->  substitute
+  0, // 27 ->  escape
+  0, // 28 ->  file separator
+  0, // 29 ->  group separator
+  0, // 30 ->  record separator
+  0, // 31 ->  unit separator
+  ' ', // 32 ->  space
+  '!', // 33 ->  !
+  0, // 34 ->  “
+  '#', // 35 ->  #
+  '$', // 36 ->  $
+  '%', // 37 ->  %
+  '&', // 38 ->  &
+  '\'', // 39 ->  ‘
+  0, // 40 ->  (
+  0, // 41 ->  )
+  '*', // 42 ->  *
+  '+', // 43 ->  +
+  0, // 44 ->  ,
+  '-', // 45 ->  –
+  '.', // 46 ->  .
+  0, // 47 ->  /
+  '0', // 48 ->  0
+  '1', // 49 ->  1
+  '2', // 50 ->  2
+  '3', // 51 ->  3
+  '4', // 52 ->  4
+  '5', // 53 ->  5
+  '6', // 54 ->  6
+  '7', // 55 ->  7
+  '8', // 56 ->  8
+  '9', // 57 ->  9
+  0, // 58 ->  :
+  0, // 59 ->  ;
+  0, // 60 ->  <
+  0, // 61 ->  =
+  0, // 62 ->  >
+  0, // 63 ->  ?
+  0, // 64 ->  @
+  'a', // 65 ->  A
+  'b', // 66 ->  B
+  'c', // 67 ->  C
+  'd', // 68 ->  D
+  'e', // 69 ->  E
+  'f', // 70 ->  F
+  'g', // 71 ->  G
+  'h', // 72 ->  H
+  'i', // 73 ->  I
+  'j', // 74 ->  J
+  'k', // 75 ->  K
+  'l', // 76 ->  L
+  'm', // 77 ->  M
+  'n', // 78 ->  N
+  'o', // 79 ->  O
+  'p', // 80 ->  P
+  'q', // 81 ->  Q
+  'r', // 82 ->  R
+  's', // 83 ->  S
+  't', // 84 ->  T
+  'u', // 85 ->  U
+  'v', // 86 ->  V
+  'w', // 87 ->  W
+  'x', // 88 ->  X
+  'y', // 89 ->  Y
+  'z', // 90 ->  Z
```

Comment:
These are all wrong and should be capitalized.

Luckily, since they are all non-zero, they are still considered valid within header names so things should continue to work.

Please add tests where you have a header name containing all valid/allowed characters and one where one of the characters is invalid (loop through each of the remaining invalid characters). We need to verify and make sure we only reject invalid header names and not valid ones.

Reply:
it's not capitalized on purpose so it can be use as toLower(uint8_t x).   

I won't expend more time adding tests for this edge case. It's already took more than it deserves.  We must wait until we are sure this is somehow worth it.

Reply:
If you are doing some trick here, then you absolutely need comments to explain this.
But upper to lower is just add 0x20 and subtract to go the other way (for chars in the A-Z range)

Reply:
> it's not capitalized on purpose so it can be use as toLower(uint8_t x).

Where do you use it as `toLower` atm? I couldn't find it. Do you foresee us using that somewhere else, in the future? Maybe we address that concern once we find a need.